plugins/amd-gpu/fu-amd-gpu-device.c

```diff
@@ -246,9 +246,19 @@ fu_amd_gpu_device_setup(FuDevice *device, GError **error)
 	self->vbios_pn = fu_strsafe((const gchar *)vbios_info.vbios_pn, PART_NUM_STR_SIZE);
 	part = g_strdup_printf("AMD\\%s", self->vbios_pn);
 	fu_device_add_instance_id(device, part);
-	fu_device_set_version_raw(device, vbios_info.version);
-	ver = fu_strsafe((const gchar *)vbios_info.vbios_ver_str, sizeof(vbios_info.vbios_ver_str));
-	fu_device_set_version(device, ver); /* nocheck:set-version */
+
+	tokens =
+	    fu_strsplit((const gchar *)vbios_info.vbios_pn, sizeof(vbios_info.vbios_pn), "-", -1);
+	if (g_strv_length(tokens) >= 3) {
+		guint64 ver;
+
+		if (!fu_strtoull(tokens[2], &ver, 0, G_MAXUINT64, FU_INTEGER_BASE_AUTO, error))
+			return FALSE;
+		fu_device_set_version_format(device, FWUPD_VERSION_FORMAT_NUMBER);
+		fu_device_set_version(device, tokens[2]); /* nocheck:set-version */
```

Comment:
nah, can you implement a `->convert_version` instead pls?

Reply:
Added